apps/heft/src/plugins/FileGlobSpecifier.ts

```diff
@@ -129,39 +136,55 @@ export async function watchGlobAsync(
   return results;
 }
 
-export async function getFilePathsAsync(
-  fileGlobSpecifier: IFileSelectionSpecifier,
-  fs?: FileSystemAdapter
-): Promise<Set<string>> {
-  const rawFiles: string[] = await glob(fileGlobSpecifier.includeGlobs!, {
-    fs,
+export async function getFileSelectionSpecifierPathsAsync(
+  options: IGetFileSelectionSpecifierPathsOptions
+): Promise<Map<string, fs.Dirent>> {
+  const { fileGlobSpecifier, includeFolders, fileSystemAdapter } = options;
+  const rawEntries: Entry[] = await glob(fileGlobSpecifier.includeGlobs!, {
+    fs: fileSystemAdapter,
     cwd: fileGlobSpecifier.sourcePath,
     ignore: fileGlobSpecifier.excludeGlobs,
     dot: true,
-    absolute: true
+    absolute: true,
+    onlyFiles: !includeFolders,
+    stats: true
   });
 
-  if (fs && isWatchFileSystemAdapter(fs)) {
-    const changedFiles: Set<string> = new Set();
+  let results: Map<string, fs.Dirent>;
+  if (fileSystemAdapter && isWatchFileSystemAdapter(fileSystemAdapter)) {
+    results = new Map();
     await Async.forEachAsync(
-      rawFiles,
-      async (file: string) => {
-        const state: IWatchedFileState = await fs.getStateAndTrackAsync(path.normalize(file));
+      rawEntries,
+      async (entry: Entry) => {
+        const { path: filePath, dirent } = entry;
+        if (entry.dirent.isDirectory()) {
```

Comment:
@dmichon-msft  does it make sense to skip directories here? Only include the watched files?

Reply:
Yes. The `readdir` calls will have already been tracked.